Makefile

```diff
@@ -380,30 +375,36 @@ create-ocp-manifests:
 	export HW_REQUIREMENTS="$(subst ",\", $(shell cat $(ROOT_DIR)/data/default_hw_requirements.json | tr -d "\n\t "))" && \
 	$(MAKE) deploy-postgres deploy-ocm-secret deploy-s3-secret deploy-service deploy-ui
 
-ci-deploy-for-subsystem: _verify_cluster generate-keys
-	export TEST_FLAGS=--subsystem-test && export AUTH_TYPE="rhsso" && export DUMMY_IGNITION=${DUMMY_IGNITION} && \
-	export IPV6_SUPPORT="True" && export ENABLE_ORG_TENANCY="True" && export ENABLE_ORG_BASED_FEATURE_GATES="True" && \
-	export RELEASE_SOURCES='$(or ${RELEASE_SOURCES},${DEFAULT_RELEASE_SOURCES})' && \
-	$(MAKE) deploy-wiremock deploy-all
+ci-deploy-for-subsystem: deploy-test
 
-patch-service: _verify_cluster update-local-image
+patch-service: _verify_cluster
+	$(MAKE) load-image
 ifdef DEBUG_SERVICE
 	$(KUBECTL) patch deployment assisted-service --type json -p='[{"op": "add", "path": "/spec/template/spec/containers/0/ports/-", "value": {"containerPort": 40000}}]'
 	$(KUBECTL) patch service assisted-service --type json -p='[{"op": "add", "path": "/spec/ports/-", value: {"name": "assisted-service-debug", "port": 40000, "protocol": "TCP", "targetPort": 40000}}]'
 	$(KUBECTL) patch deployment assisted-service --type json -p='[{"op": "remove", "path": "/spec/template/spec/containers/0/livenessProbe"}]'
 endif
 	$(call restart_service_pods)
 
-deploy-test: _verify_cluster generate-keys update-local-image
+deploy-test: _verify_cluster generate-keys
 	-$(KUBECTL) delete deployments.apps assisted-service &> /dev/null
-	export SERVICE=${LOCAL_SERVICE_IMAGE} && export TEST_FLAGS=--subsystem-test && \
+	export TEST_FLAGS=--subsystem-test && \
 	export AUTH_TYPE="rhsso" && export DUMMY_IGNITION="True" && \
 	export IPV6_SUPPORT="True" && ENABLE_ORG_TENANCY="True" && ENABLE_ORG_BASED_FEATURE_GATES="True" && \
 	export RELEASE_SOURCES='$(or ${RELEASE_SOURCES},${DEFAULT_RELEASE_SOURCES})' && \
 	$(MAKE) deploy-wiremock deploy-all
 
-# An alias for the deploy-test target
-deploy-service-for-subsystem-test: deploy-test
+deploy-service-for-subsystem-test:
+	$(MAKE) create-hub-cluster
```

Comment:
why `$(MAKE)` and not used as a dependency?

Reply:
I wanted to control the order of executions. to my understanding when you specify list of dependencies e.g:

```
deploy-service-for-subsystem-test: create-hub-cluster load-image
```

there is no guarantee that this is the order it will execute unless one of the dependencies is also a dependency of the other, which shouldn't be the case here

Reply:
Dependencies are executed in order, unless further dependencies state otherwise. If you really need one step to be executed before another, you can define so in the same step:
```
deploy-service-for-subsystem-test: create-hub-cluster load-image
    ...

load-image: create-hub-cluster
   ...
```

Reply:
Done